### ISSUE TITLE
[accessibility-defect-2]: Checkbox group Hint text is not read by screen readers

### ISSUE THREAD

**Issue by jeana-adhoc** [department-of-veterans-affairs/va.gov-team#71634]
### Point of contact

@jeana-adhoc

### Severity level

2, Serious. Should be fixed in 1-2 sprints post-launch.

### Details

On the [records selection page](https://staging.va.gov/records/request-personal-records-form-20-10206/record-selections), there is a checkbox group with hint text. The hint text is loading outside of the legend and so it is not being read by the screen reader. 

### Reproduction steps

1. Enable voiceover or another screen reader
2. Go to the [records selection page](https://staging.va.gov/records/request-personal-records-form-20-10206/record-selections) page
3. Navigate into the checkbox group. 
4. You will never hear the hint text announced.

### Proposed solution or next steps

When looking at the [checkbox group in the design system](https://design.va.gov/storybook/?path=/docs/uswds-va-checkbox-group--default) and adding hint text, I see that it is loading the hint text in the legend as expected.  Our component looks like it should be the v3 version, but something is not quite right.

| Storybook/V3      | VFF Team implementation |
| ----------- | ----------- |
| <img width="564" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/06828fcb-27ec-480f-8953-e3844edd7931"> | <img width="564" alt="image" src="https://github.com/department-of-veterans-affairs/va.gov-team/assets/117098918/a1e19601-075e-4106-9f2d-ca05741b23e4"> |

### References, articles, or WCAG support

This almost falls under VA.gov Experience Standard 09.05 (Visual labels aren't included in the accessible label), which would be a launch-blocking issue. But since it's helper text and not the label itself. Logging it as other/not launch blocking. (Following other guidance from @briandeconinck )

- [WCAG SC 2.5.3_A](https://www.w3.org/WAI/WCAG21/quickref/#label-in-name)

### Type of issue

- [ ] Axe-core
- [X] Screenreader | assistive tech & device support
- [ ] Keyboard
- [ ] Zoom
- [ ] Color, typography, & visual elements
- [ ] Components and pattern standards
- [ ] Content organization
- [ ] Focus
- [ ] Headings
- [ ] Content style guide
- [ ] Link and button labels
- [ ] Semantic HTML
- [ ] Markup and metadata
- [ ] Non-text content and media
- [ ] Overall user experience
- [X] Cognitive | user flows & navigating
- [ ] URL standards
- [ ] Something else

**Comment by tlei123** [comment]
Closing this unfixed.  Turns out this is a _design-component_ bug.  va-checkbox-group is rendering the hint _outside_ the <legend/> tag.

![image](https://github.com/department-of-veterans-affairs/va.gov-team/assets/587583/12e43e5c-66f5-458c-8ad8-aa730400b623)

**Comment by jeana-adhoc** [comment]
[I have filed this ticket with the design system](https://github.com/department-of-veterans-affairs/vets-design-system-documentation/issues/2342)
